[dynamo] save/restore system random state more carefully

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #145750

Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

williamwen42 committed:
[dynamo] restore system random state more carefully

[ghstack-poisoned]

williamwen42 commented:
@williamwen42 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D68723334).

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68723334](https://our.internmc.facebook.com/intern/diff/D68723334)

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge

huydhn commented:
@williamwen42 There is a smoke perf benchmark failure after this land.  Do you know if it's related to the change?

[GH job link](https://github.com/pytorch/pytorch/actions/runs/13001859947/job/36262537806) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/e3d3f2b22e4b75c64eaa2f940a2dd80c1e43435c)

williamwen42 commented:
I don't expect a perf regression from this PR.

eellison commented:
@pytorchbot revert

eellison commented:
@pytorchbot revert -m "bisected perf regression" -c nosignal

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68723334](https://our.internmc.facebook.com/intern/diff/D68723334)

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68723334](https://our.internmc.facebook.com/intern/diff/D68723334)

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68723334](https://our.internmc.facebook.com/intern/diff/D68723334)

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68723334](https://our.internmc.facebook.com/intern/diff/D68723334)

[ghstack-poisoned]

williamwen42 commented:
@williamwen42 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D68918404).

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

[ghstack-poisoned]

jansel commented:
Re-request review once you have figured out the regression issue.

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

[ghstack-poisoned]

williamwen42 commented:
I removed the random state preservation from `eval_frame.py`, which lower overhead on the hotpath.

I can move more of `preserve_global_state` into `eval_frame.c` in a followup.

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

[ghstack-poisoned]

jansel reviewed (changes requested):
Test failures?


Can you post a screenshot of the result of a benchmark run?

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo] save/restore system random state more carefully"


Reattempt of https://github.com/pytorch/pytorch/pull/145435 since the state of the linked internal diff appears to be messed up.

Note: I have verified that the previously failing internal tests now pass internally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Differential Revision: [D68918404](https://our.internmc.facebook.com/intern/diff/D68918404)

[ghstack-poisoned]

williamwen42 commented:
Closing as there is a reattempt PR
